### ISSUE TITLE
Costura doesn't embed resource satellite assemblies

### ISSUE THREAD

**Issue by dguder** [Fody/Costura#61]
I have some issues regarding resource assemblies in fody.costura. It could be that I'm doing something wrong. 

When I add some localization resources they are not merged in correctly.
I added two projects to my fork of [FodyAddinSamples](https://github.com/dguder/FodyAddinSamples/tree/costura-resources) to demonstrate the issue.
Both projects are working if the resources exists in their according folders. But if I delete the language folders (as seen in `TestFixtureSetUp`) then the localization is not working anymore. 
For the internal resources the detailed log showed that Fody runs before satellite assemblies are created. And for the external resources (my most important use case) I have no idea what happens.

Maybe the bugs on MS side, as stated in this comment at [Issue #52](https://github.com/Fody/Costura/issues/52#issuecomment-35053028), are the reason for this. 

IMHO the main point is that the satellite assemblies are not found or still not exist at compile-time when Fody is called.
Your internal test is working, since the resource file is added manually by code and not searched and added by Fody's msbuild target.

One point I want to mention is that everything is fine when I add the resource as reference. But then I can only add a single translation dll. I can add an example for this as well, if it makes sense.

I'm willing to provide a pull request as fix, but I have no glue where ti start right now.

So far and many thanks for Costura-Fody.
Dominik

**Comment by MarcStan** [comment]
While the proposed solution by olsh works for Winforms, it doesn't for WPF as there is no designer code.

I always wanted to use code generation, so I cooked up [Resource Embedder](https://github.com/MarcStan/Resource.Embedder) ([NuGet](https://www.nuget.org/packages/Resource.Embedder/)).

It's heavily inspired by Costura and works the same way but runs directly after the satellite assemblies are generated thus allowing to embed them.

**Comment by wintersnare** [comment]
@MarcStan I was just looking for this exact functionality today, due to the exact reasons you stated. Thanks for this, I'll definitely be checking out your Resource Embedder.

**Comment by DoomMuffins** [comment]
Hey, I just stumbled upon a very similar problem myself when trying to embed an SGen-generated XmlSerializers dll (A pre-generated MyTarget.XmlSerializers.dll in the output directory). I cannot add a reference to it because that would give me a build error, but I'd like to explicitly embed it. Could a way to force-embed non-referenced assemblies be added for such corner cases?
I actually saw a workaround in the google code archive of the previous repository in the following link, but I'd love a cleaner/documented way to achieve this.
https://code.google.com/p/costura/source/browse/EmbedWithNoReference.wiki?repo=wiki&r=a87656f4810ab169570df7811fe7d05b69e48b95
